### ISSUE TITLE
event ends next day

### ISSUE THREAD

**Issue by hypertextdevelopers** [mattlewis92/angular-calendar#1234]
Hi, the case is that, when the dragged event reaches the bottom of calendar it cannot be moved lower and due to that we are unable to move the event through the night. For example we have event with duration of 4 hours, we wanted to have this event started at 23:00 and ended at 03:00. Due to that event drag stops at the bottom of the calendar we cannot set apparent event's start and end. The only thing which is done is that the moment the event picks the bottom it gets a flag of "ends after day" set to true and shows itself on the next day, but we can only move this second part of event which is on the next day. We cannot move down the part where event starts. This behaviour is weird.

It should behaves like while dragging our event reaches the bottom it arises on the next day and we can drag down more and more this event but on the day it starts, and by that this part of event on the next day also is being dragged down.

**Screenshots**

![image](https://user-images.githubusercontent.com/45067921/79458499-2e128400-7ff2-11ea-8a81-51d31ea8ce9b.png)
![image](https://user-images.githubusercontent.com/45067921/79458529-379bec00-7ff2-11ea-8113-d947e4203273.png)

"@angular/core": "~8.2.14",
"angular-calendar": "0.27.14",
Version 1.7.92 Chromium: 80.0.3987.163 (Official Build) (64-bit)

**Comment by mattlewis92** [comment]
Thanks for opening an issue, the current behaviour is indeed a bit strange, especially as resizing events does behave this way but dropping events doesn't, they should really both be consistent. I will get this fixed in the next version!

**Comment by mattlewis92** [comment]
Hey there, I just released a beta of `0.29` with this fix. You can find instructions on how to update to it here: https://github.com/mattlewis92/angular-calendar/pull/1249 Please can you give it a try and let me know if it works for you, thanks! 😄

**Comment by hypertextdevelopers** [comment]
Hey, I am not able right now to built project because I was exploiting a lot of DayView functions and so on but right now there is a lack of them.
For example getDayView(), but right now it is not available in CalendarUtils

**Comment by mattlewis92** [comment]
Ah right, that's because of the merging of the week and day view. You can see the commit with instructions on how to migrate: https://github.com/mattlewis92/calendar-utils/commit/a23983f8d58e98a2e13e0f0060da8b15de3ff455

**Comment by hypertextdevelopers** [comment]
Hi, before update I was extending your component CalendarDayViewComponent, which had something like "view" object, right now it is not available in this component and I am confused with that. Do you have any suggestions along this case?
The same case for all of methods that this component had and right now it isn't having.

**Comment by mattlewis92** [comment]
It's because the day view and the week view are now the same thing (a day view is just a week view with `[daysInWeek]="1"`), if you have a look at the source you'll see: https://github.com/mattlewis92/angular-calendar/blob/master/projects/angular-calendar/src/modules/day/calendar-day-view.component.ts#L31

So you want to extend the `CalendarWeekViewComponent` instead and set `daysInWeek = 1` on the component to get the same behaviour as the old day view.

**Comment by hypertextdevelopers** [comment]
okay so as far as I understand the way to show events on daily calendar is the same like on week?

**Comment by mattlewis92** [comment]
Yep, they're now the exact same code. You might find it helpful to see the [breaking changes list](https://github.com/mattlewis92/angular-calendar/blob/master/CHANGELOG.md#0280-2019-11-14)

**Comment by hypertextdevelopers** [comment]
Hi everything works, great job. But I feel like that there is one more bug I mean that when we exploit mwlClick anywhere for instance on calendar we cannot go along with document click.

My scenerio, I have a context menu on right click that is closed on left click triggered on document. But the time we use mwlClick this click on document doesn't trigger, in previous version 0.27.14 everything was working fine.

**Comment by hypertextdevelopers** [comment]
The same with dayHeaderClicked

**Comment by mattlewis92** [comment]
I'm glad to hear everything it working ok 😄 

In regards to your issue, it's probably because of the `stopPropagation` event that's emitted: https://github.com/mattlewis92/angular-calendar/blob/master/projects/angular-calendar/src/modules/common/click.directive.ts#L38

That click directive used to do a lot more than it does now, I would recommend just replacing it with a regular `(click)` event (as really all it does now is call `stopPropagation` which is causing your problem)

**Comment by hypertextdevelopers** [comment]
One more thing is there a possiblity to prevent events to ending the next day?

**Comment by mattlewis92** [comment]
Sorry I'm not quite sure I understand what you mean?

**Comment by hypertextdevelopers** [comment]
I apologise, I have explained it extremly bad.
I mean that I would like to know if there is a possiblity to disable dividing event into two parts when it ends after day. For example, we have event that starts at 11 pm and it lasts 3 hours so it ends up the next day, and in the calendar event is divided into two parts one on start day and second on end day. I would like to disable it and have the show in one piece with full length even if the show ends the next day.

**Comment by mattlewis92** [comment]
You could adjust the end date of events to not expand onto multiple days, something like:

```
events = events.map(event => {
  if (!isSameDay(startOfDay(event.start), startOfDay(event.end))) {
    return {
       ...event,
       end: endOfDay(event.start)
    }
  }
  return event
})
```

**Comment by stefanomiccoli** [comment]
not sure if also my issue will be closed with 0.29.0, however im experiencing a eird inconsistency in the rendering of events.

One pic worth 100 words.
![image](https://user-images.githubusercontent.com/1469822/95433302-0de3d200-0950-11eb-960d-44084801b5fa.png)

what is happening is basically that event with dates
```
{
  "start": "2020-10-07T22:00:00.000Z",
  "end": "2020-10-08T00:00:00.000Z"
}
```
is rendered as expected within the segments of the the hours 12am and 1am, without using the 2am (GMT+2 here).

when the event ends on midnight however, also segment of next day is used too. I know this could be addressed similarly to what matt suggests in the last, but of course it would make a special case id prefer to avoid for the sake of maintenability.
